spread.yaml

```diff
@@ -55,6 +55,29 @@ backends:
                 username: ubuntu
                 password: ubuntu
 
+    adhoc-ubuntu-core:
+        type: adhoc
+        environment:
+            SPREAD_ADHOC_UC_ADDRESS: "$(HOST: echo ${SPREAD_ADHOC_UC_ADDRESS:-localhost})"
+            SPREAD_ADHOC_UC_PORT: "$(HOST: echo ${SPREAD_ADHOC_UC_PORT:-8022})"
+        allocate: |
+            echo "Allocating $SPREAD_SYSTEM..."
+            ADDRESS $SPREAD_ADHOC_UC_ADDRESS:$SPREAD_ADHOC_UC_PORT
+        discard: |
+            echo "Discarding ad-hoc $SPREAD_SYSTEM"
+        restore: |
+            rm -rf $GOPATH/src
+        systems:
+            - ubuntu-core-16.04-64:
+                username: test
+                password: ubuntu
+            - ubuntu-core-16-device-amd64:
```

Comment:
As we discussed, it's best to support these variations via a different mechanism instead of different systems. That's the sort of reason why variants have been created, for example.